packages/grafana-ui/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { PureComponent, useContext, FC } from 'react';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { ThemeContext } from '../../themes/index';
+import { css, cx } from 'emotion';
+import { GrafanaTheme } from '../../types/theme';
+import tinycolor from 'tinycolor2';
+import { ToggleButtonGroupProps, ToggleButtonProps, ToggleButtonState } from './types';
+import { increaseContrast } from '../../utils/colors';
+import { ButtonSize } from '../Button/AbstractButton';
+
+const ToggleButtonGroup: FC<ToggleButtonGroupProps> = ({ label, children, propsPriority }) => {
+  const styles = getStyles(useContext(ThemeContext));
+  return (
+    <div>
+      <span className={styles.label}>{label}</span>
+      {React.Children.map(children, (child: any, index) => {
+        // Set correct style depending on first/middle/last
+        const style = index === 0 ? styles.first : index === children.length - 1 ? styles.last : styles.middle;
+
+        return React.cloneElement(child, {
+          className: style,
+          propsPriority,
+        });
+      })}
+    </div>
+  );
+};
+
+ToggleButtonGroup.displayName = 'ToggleButtonGroup';
+
+export { ToggleButtonGroup };
+export class ToggleButton extends PureComponent<ToggleButtonProps, ToggleButtonState> {
```

Comment:
Does ToggleButton follow the same sizing as regular buttons(I think it should)? Then we could use `AbstractButton` component for rendering ToggleButton

Reply:
If you ask me, I think the sizing for regular buttons are varying too much. Wouldn't make sense if `ToggleButton` is meant to be used as a form component.
![Screenshot from 2019-07-08 15-35-17](https://user-images.githubusercontent.com/1438972/60814651-48335200-a196-11e9-9b2a-882bb5a3a58a.png)